src/ServiceStack/DependencyInjection/DependencyService.cs

```diff
@@ -37,22 +39,82 @@ public DependencyService()
             _containerBuilder = new ContainerBuilder();
         }
 
-        public void RegisterTypeAsItself(Type classType, Sharing sharing = Sharing.None)
+        public void Register(Type implementingType, Sharing sharing, bool registerAsImplementedInterfaces, bool includeNonPublicConstructors)
         {
-            var registration = _containerBuilder.RegisterType(classType);
-            SetSharing(registration, sharing);
+            if (implementingType.IsGenericType)
+            {
+                var registration = _containerBuilder.RegisterGeneric(implementingType);
+                if (registerAsImplementedInterfaces)
+                {
+                    registration = registration.AsImplementedInterfaces();
+                }
+                if (includeNonPublicConstructors)
```

Comment:
Is this standard formatting for stacked `if` statements? Initially I read this as an `if / else`. Consider adding a newline between them